Tools/paletteable sprite editor/js/marvinj-0.9.js

```diff
@@ -0,0 +1,2665 @@
+};
+
+MarvinImage.prototype.getIntComponent1 = function(x,y){
+	var start = ((y*this.getWidth())+x)*4; 
+	return this.imageData.data[start+1];
+};
+
+MarvinImage.prototype.getIntComponent2 = function(x,y){
+	var start = ((y*this.getWidth())+x)*4; 
+	return this.imageData.data[start+2];
+};
+
+MarvinImage.prototype.setIntColor = function(x,y, a1, a2, a3, a4){
+	if(a2 == null){
+		this.setIntColor1(x,y,a1);
+	} else if(a3 == null && a4 == null){ 
+		this.setIntColor2(x,y,a1,a2);
+	}
+	else if(a4 == null){
+		this.setIntColor3(x,y,a1,a2,a3);
+	}
+	else{
+		this.setIntColor4(x,y,a1,a2,a3,a4);
+	}
+};
+
+MarvinImage.prototype.getIntColor = function(x,y){
+	var start = ((y*this.getWidth())+x)*4;
+	
+	return 	0x100000000 + 
+			(this.imageData.data[start+3] << 24) + 
+			(this.imageData.data[start] << 16) +
+			(this.imageData.data[start+1] << 8) +
+			(this.imageData.data[start+2]);
+};
+
+MarvinImage.prototype.setIntColor1 = function(x,y, color){
+	var a = (color & 0xFF000000) >>> 24;
+	var r = (color & 0x00FF0000) >> 16;
+	var g = (color & 0x0000FF00) >> 8;
+	var b = color & 0x000000FF;
+	this.setIntColor4(x,y,a,r,g,b);
+};
+
+MarvinImage.prototype.setBinaryColor = function(x,y,value){
+	var pos = ((y*this.getWidth())+x);
+	this.arrBinaryColor[pos] = value;
+};
+
+MarvinImage.prototype.getBinaryColor = function(x,y){
+	var pos = ((y*this.getWidth())+x);
+	return this.arrBinaryColor[pos];
+};
+
+MarvinImage.copyColorArray = function(imgSource, imgDestine){
+		
+	if(imgSource.getColorModel() == imgDestine.getColorModel()){
+		switch(imgSource.getColorModel()){
+			case MarvinImage.COLOR_MODEL_RGB:
+				for(var i=0; i<imgSource.imageData.data.length; i++){
+					imgDestine.imageData.data[i] = imgSource.imageData.data[i];
+				}
+				break;
+			case MarvinImage.COLOR_MODEL_BINARY:
+				for(var i=0; i<imgSource.arrBinaryColor.length; i++){
+					imgDestine.arrBinaryColor[i] = imgSource.arrBinaryColor[i];
+				}
+				break;
+		}
+	}
+};
+
+MarvinImage.prototype.drawRect = function(x,y, width, height, color){
+	for(var i=x; i<x+width; i++){
+		this.setIntColor(i, y, color);
+		this.setIntColor(i, y+(height-1), color);
+	}
+	
+	for(var i=y; i<y+height; i++){
+		this.setIntColor(x, i, color);
+		this.setIntColor(x+(width-1), i, color);
+	}
+};
+
+MarvinImage.prototype.fillRect = function(x,y, width, height, color){
+	for(var i=x; i<x+width; i++){
+		for(var j=y; j<y+height; j++){
+			if(i < this.getWidth() && j < this.getHeight()){
+				this.setIntColor(i,j,color);
+			}
+		}
+	}
+};
+
+MarvinImage.prototype.setColorToAlpha = function(alpha, color){
+	for(var y=0; y<this.height; y++){
+		for(var x=0; x<this.width; x++){
+			if((this.getIntColor(x,y) & 0x00FFFFFF) == (color & 0x00FFFFFF)){
+				this.setAlphaComponent(x,y,alpha);
+			}
+		}
+	}
+};
+
+MarvinImage.prototype.setAlphaToColor = function(color){
+	for(var y=0; y<this.height; y++){
+		for(var x=0; x<this.width; x++){
+			if(this.getAlphaComponent(x, y) == 0){
+				this.setIntColor(x, y, 0xFFFFFFFF);
```

Comment:
Codacy found an issue: [The numeric literal '0xFFFFFFFF' will have at different value at runtime.](https://app.codacy.com/gh/unitystation/unitystation/pullRequest?prid=6395467)